File: 1721-swapping-nodes-in-a-linked-list/1721-swapping-nodes-in-a-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* swapNodes(ListNode* head, int k) {
        int i = 1,n = 1;
        ListNode * curr = head;
        ListNode * first = new ListNode(),*second = new ListNode();
        while(curr!=NULL){
            curr = curr->next;
            n++;
        }
        curr = head;
        while(curr!=NULL){
            if(i == k){
                first = curr;
            }
            if(n-i == k){
                second = curr;
            }
            curr = curr->next;
            i++;
        }
        int temp = first->val;
        first->val = second->val;
        second->val = temp;
        return head;
        
    }
};

File: 1721-swapping-nodes-in-a-linked-list/1721-swapping-nodes-in-a-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode swapNodes(ListNode head, int k) {
        int n = 1;
        ListNode curr = head;
        while(curr!=null){
            n++;
            curr = curr.next;
        }
        curr = head;
        ListNode start = new ListNode();
        ListNode end = new ListNode();
        int i = 1;
        while(curr!=null){
            if(i == k){
                start = curr;
            }
            if(k == n-i){
                end = curr;
            }
            curr = curr.next;
            i++;
        }
        int temp = start.val;
        start.val = end.val;
        end.val = temp;
        return head;
    }
}